src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs

```diff
@@ -422,7 +422,12 @@ private void LoadAndInitialize(DataCollectorSettings dataCollectorSettings, stri
                 // Look up the extension and initialize it if one is found.
                 var extensionManager = this.DataCollectorExtensionManager;
                 var dataCollectorUri = string.Empty;
-                this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri);
+
+                if (!this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri))
```

Comment:
Shouldn't we do this when datacollectorUri is not specified by the user ?
If user already specified it, and that's what we need, why are trying to fetch it again ?